lib/datasource/go/index.ts

```diff
@@ -68,12 +69,24 @@ async function getDatasource(goModule: string): Promise<DataSource | null> {
           .replace(/\/$/, ''),
       };
     }
-    const gitlabRes = gitlabRegExp.exec(goSourceUrl);
-    if (gitlabRes) {
+    const gitlabUrl = gitlabHttpsRegExp.exec(goSourceUrl);
+    const gitlabModule = gitlabRegExp.exec(goModule);
+    const gitlabUrlName = gitlabUrl?.[2].replace(/\/$/, '');
+    const gitlabModuleName = gitlabModule?.[2].replace(/\/$/, '');
+
+    if (gitlabModuleName?.startsWith(gitlabUrlName)) {
       return {
         datasource: gitlab.id,
-        registryUrl: gitlabRes[1],
-        lookupName: gitlabRes[2].replace(/\/$/, ''),
+        registryUrl: gitlabUrl[1],
+        lookupName: gitlabModuleName,
+      };
+    }
+
+    if (gitlabUrl) {
+      return {
+        datasource: gitlab.id,
+        registryUrl: gitlabUrl[1],
+        lookupName: gitlabUrlName,
       };
     }
```

Comment:
If just the `lookupName` differs then these two if statements could be de-duplicated